### ISSUE TITLE
Gmail SMTP timeout, MS Live false negative

### ISSUE THREAD

**Issue by jasonbjorn01** [sharsil/mailcat#7]
I've tested some valid Gmail and Hotmail/Outlook emails in Mailcat, but none of them returns as valid emails in Mailcat.

**Comment by soxoj** [comment]
Hey, @jasonbjorn01! 
The reason could be in rate limiting or your IP ban.
Please, give examples of such accounts, I'll check it.

**Comment by jasonbjorn01** [comment]
Here's two random emails I've verified in Skype: patrickwhite1985@gmail.com and patrickwhite@hotmail.com. None of them return as valid Gmail or Hotmail in Mailcat. I've changed my IP and run new checks, but still nothing. 
Is it possible to add e.g: "Gmail - Blocked request" - if the IP has been flaged?

**Comment by jasonbjorn01** [comment]
Like we get in Holehe.

**Comment by soxoj** [comment]
I have checked and got the following results: gmail.com exists, hotmail.com is absent (but I see it is already registered with the manual check).

I think there are two different problems:
- SMTP ban for Gmail check
- False-positive for MS Live check

Sure, it is possible to display errors and recommendations to restart failed checks with a proxy. We've already thought about it and are planning improvement.
Also, I added a task to research and improve MS Live check.

You can track tasks [here](https://github.com/sharsil/mailcat/projects/1), also I write you in this thread after fixes.

Thank you for your feedback!

**Comment by jasonbjorn01** [comment]
Thanks!

**Comment by soxoj** [comment]
I've added errors displaying for SMTP checks, please, try to check Gmail now with the actual version of Mailcat.

**Comment by jasonbjorn01** [comment]
This is what I get when I run the hotmail above;
Error while checking gmail: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond Error while checking yandex: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond

mail.de:
Traceback (most recent call last):
  File "C:\Users\cmder\mailcat\mailcat.py", line 1507, in <module>
    asyncio.get_event_loop().run_until_complete(jobs)
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 642, in run_until_complete
    return future.result()
  File "C:\Users\cmder\mailcat\mailcat.py", line 1383, in print_results
    res = await checker(target, req_session_fun)
  File "C:\Users\cmder\mailcat\mailcat.py", line 1084, in iCloud
    check = await sreq.post('https://iforgot.apple.com/password/verify/appleid', headers=headers, data=json.dumps(data), allow_redirects=False, timeout=5)
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\site-packages\aiohttp\client.py", line 634, in _request
    break
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\site-packages\aiohttp\helpers.py", line 721, in __exit__
    raise asyncio.TimeoutError from None
asyncio.exceptions.TimeoutError

**Comment by jasonbjorn01** [comment]
And I get the same response for the Gmail

**Comment by soxoj** [comment]
Well, looks like a connectivity issue, moreover, it affects SMTP check for Yandex also.
Please, run exactly the following commands and send their output.
```
python3 mailcat.py -p gmail  patrickwhite1985 -svd
python3 mailcat.py -p gmail  patrickwhite1985 -svd --tor
```

**Comment by jasonbjorn01** [comment]
python3 mailcat.py -p gmail  patrickwhite1985 -svd
Running gmail checker for patrickwhite1985...
[WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
Traceback (most recent call last):
  File "C:\Users\cmder\mailcat\mailcat.py", line 78, in code250
    server.connect(mxRecord)
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\smtplib.py", line 341, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\smtplib.py", line 312, in _get_socket
    return socket.create_connection((host, port), timeout,
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\socket.py", line 843, in create_connection
    raise err
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\socket.py", line 831, in create_connection
    sock.connect(sa)
TimeoutError: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
No results for gmail
Error while checking gmail: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond

**Comment by jasonbjorn01** [comment]
python3 mailcat.py -p gmail  patrickwhite1985 -svd --tor
Using tor to make requests...
Running gmail checker for patrickwhite1985...
[WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
Traceback (most recent call last):
  File "C:\Users\cmder\mailcat\mailcat.py", line 78, in code250
    server.connect(mxRecord)
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\smtplib.py", line 341, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\smtplib.py", line 312, in _get_socket
    return socket.create_connection((host, port), timeout,
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\socket.py", line 843, in create_connection
    raise err
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\socket.py", line 831, in create_connection
    sock.connect(sa)
TimeoutError: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
No results for gmail
Error while checking gmail: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond

**Comment by soxoj** [comment]
Got it. I think it might be caused by your firewall or ISP restrictions.
Please, check the connectivity to port 25 of email provider manually, you can use telnet so:
```
telnet alt4.gmail-smtp-in.l.google.com. 25
```
If there is no message `Connected to ...`, then try to disable your antiviruses and other similar software, and check again. Also, you can check if your IP is blacklisted by ISPs here: https://www.dnsbl.info. You can read instructions for removal IP from blacklist for each of the sources.

Also, I realized, we have no support of Tor for SMTP :( 
I'll make it and will return to you 👍

**Comment by jasonbjorn01** [comment]
Hmm - Disabled Firewall, Anti-virus and the VPN, but still the same error response. I'll check on another pc..

**Comment by jasonbjorn01** [comment]
I just tested on another pc and it worked fine several times. Then I deleted my mailcat folder and cloned a new one. Now I get the same errors on that pc as well. Are you sure there's nothing wrong on your side?

**Comment by jasonbjorn01** [comment]
Deleted the latest clone and replaced it with the old one. Now it works again..

**Comment by soxoj** [comment]
> Are you sure there's nothing wrong on your side?

Please, attach the outputs of both versions.

**Comment by jasonbjorn01** [comment]
And this is the error when I try to search with --tor with the oldest version:
λ mailcat.py patrickwhite1985 --tor

                  ,-.                    ^
                 ( (        _,---._ __  / \
                  ) )    .-'       `./ /   \
                 ( (   ,'            `/    /:
                  \ `-"             \'\   / |
                   .              ,  \ \ /  |
                   / @          ,'-`----Y   |
                  (            ;        :   :
                  |  .-.   _,-'         |  /
                  |  | (  (             | /
                  )  (  \  `.___________:/
                  `..'   `--' :mailcat:

Exception in thread Thread-20Exception in thread Thread-21:
:
Traceback (most recent call last):
Traceback (most recent call last):
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\socks.py", line 787, in connect
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\socks.py", line 787, in connect
        super(socksocket, self).connect(proxy_addr)super(socksocket, self).connect(proxy_addr)

ConnectionRefusedErrorConnectionRefusedError: : [WinError 10061] No connection could be made because the target machine actively refused it[WinError 10061] No connection could be made because the target machine actively refused it


During handling of the above exception, another exception occurred:


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
Traceback (most recent call last):
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\contrib\socks.py", line 96, in _new_conn
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\contrib\socks.py", line 96, in _new_conn
    conn = socks.create_connection(
      File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\socks.py", line 209, in create_connection
conn = socks.create_connection(
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\socks.py", line 209, in create_connection
    raise err
      File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\socks.py", line 199, in create_connection
raise err
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\socks.py", line 199, in create_connection
    sock.connect((remote_host, remote_port))
      File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\socks.py", line 47, in wrapper
sock.connect((remote_host, remote_port))
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\socks.py", line 47, in wrapper
    return function(*args, **kwargs)
      File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\socks.py", line 800, in connect
return function(*args, **kwargs)
      File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\socks.py", line 800, in connect
raise ProxyConnectionError(msg, error)
socks.    ProxyConnectionErrorraise ProxyConnectionError(msg, error):
Error connecting to SOCKS5 proxy 127.0.0.1:9050: [WinError 10061] No connection could be made because the target machine actively refused itsocks
.
During handling of the above exception, another exception occurred:

ProxyConnectionErrorTraceback (most recent call last):
:   File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\connectionpool.py", line 699, in urlopen
Error connecting to SOCKS5 proxy 127.0.0.1:9050: [WinError 10061] No connection could be made because the target machine actively refused it

During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
httplib_response = self._make_request(  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\connectionpool.py", line 699, in urlopen

  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\connectionpool.py", line 382, in _make_request
        httplib_response = self._make_request(self._validate_conn(conn)

  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\connectionpool.py", line 382, in _make_request
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\connectionpool.py", line 1010, in _validate_conn
    self._validate_conn(conn)
      File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\connectionpool.py", line 1010, in _validate_conn
conn.connect()
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\connection.py", line 353, in connect
        conn = self._new_conn()conn.connect()

  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\contrib\socks.py", line 127, in _new_conn
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\connection.py", line 353, in connect
    raise NewConnectionError(
conn = self._new_conn()urllib3.exceptions
.  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\contrib\socks.py", line 127, in _new_conn
NewConnectionError: <urllib3.contrib.socks.SOCKSHTTPSConnection object at 0x000001607FD60700>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it
raise NewConnectionError(
During handling of the above exception, another exception occurred:


Traceback (most recent call last):
urllib3.exceptions  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\adapters.py", line 439, in send
.NewConnectionError: <urllib3.contrib.socks.SOCKSHTTPSConnection object at 0x000001607FD6C490>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it
resp = conn.urlopen(
During handling of the above exception, another exception occurred:


Traceback (most recent call last):
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\connectionpool.py", line 755, in urlopen
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\adapters.py", line 439, in send
    retries = retries.increment(
resp = conn.urlopen(  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\util\retry.py", line 574, in increment

  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\connectionpool.py", line 755, in urlopen
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
    urllib3.exceptionsretries = retries.increment(.
MaxRetryError  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\urllib3\util\retry.py", line 574, in increment
: SOCKSHTTPSConnectionPool(host='iforgot.apple.com', port=443): Max retries exceeded with url: /password/verify/appleid (Caused by NewConnectionError('<urllib3.contrib.socks.SOCKSHTTPSConnection object at 0x000001607FD60700>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it'))
raise MaxRetryError(_pool, url, error or ResponseError(cause))
During handling of the above exception, another exception occurred:


Traceback (most recent call last):
urllib3.exceptions  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\threading.py", line 954, in _bootstrap_inner
.MaxRetryError: SOCKSHTTPSConnectionPool(host='runbox.com', port=443): Max retries exceeded with url: /signup/signup (Caused by NewConnectionError('<urllib3.contrib.socks.SOCKSHTTPSConnection object at 0x000001607FD6C490>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it'))
self.run()
During handling of the above exception, another exception occurred:


Traceback (most recent call last):
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\threading.py", line 892, in run
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\threading.py", line 954, in _bootstrap_inner
    self._target(*self._args, **self._kwargs)
      File "C:\Users\mailcat\mailcat.py", line 1198, in print_results
self.run()
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\threading.py", line 892, in run
    res = checker(target, req_session_fun)
      File "C:\Users\mailcat\mailcat.py", line 1046, in iCloud
self._target(*self._args, **self._kwargs)
  File "C:\Users\mailcat\mailcat.py", line 1198, in print_results
    check = sreq.post('https://iforgot.apple.com/password/verify/appleid', headers=headers, data=json.dumps(data), allow_redirects=False, timeout=5)
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\sessions.py", line 590, in post
    res = checker(target, req_session_fun)
      File "C:\Users\mailcat\mailcat.py", line 1013, in runbox
return self.request('POST', url, data=data, json=json, **kwargs)
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\sessions.py", line 542, in request
    chkRunbox = sreq.post('https://runbox.com/signup/signup', headers=headers, data=data, timeout=5)
      File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\sessions.py", line 590, in post
resp = self.send(prep, **send_kwargs)
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\sessions.py", line 655, in send
    return self.request('POST', url, data=data, json=json, **kwargs)
      File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\sessions.py", line 542, in request
r = adapter.send(request, **kwargs)
  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\adapters.py", line 516, in send
    resp = self.send(prep, **send_kwargs)
      File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\sessions.py", line 655, in send
raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError:     SOCKSHTTPSConnectionPool(host='iforgot.apple.com', port=443): Max retries exceeded with url: /password/verify/appleid (Caused by NewConnectionError('<urllib3.contrib.socks.SOCKSHTTPSConnection object at 0x000001607FD60700>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it'))r = adapter.send(request, **kwargs)

  File "C:\Users\cmder\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: SOCKSHTTPSConnectionPool(host='runbox.com', port=443): Max retries exceeded with url: /signup/signup (Caused by NewConnectionError('<urllib3.contrib.socks.SOCKSHTTPSConnection object at 0x000001607FD6C490>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it'))

**Comment by soxoj** [comment]
Great! Got it, different resource network utilization for old multi-threaded requests and new async requests. I'll make some optimizations.

Meanwhile, you can try to increase the timeout for Apple manually [in line 1084](https://github.com/sharsil/mailcat/blob/main/mailcat.py#L1084).

**Comment by soxoj** [comment]
Please, check [this version](https://github.com/sharsil/mailcat/archive/refs/heads/proxy-timeouts-errors.zip), I've added a timeout for STMP.

**Comment by jasonbjorn01** [comment]
C:\Users\cmder\mailcat-proxy-timeouts-errors
λ mailcat.py test

                  ,-.                    ^
                 ( (        _,---._ __  / \
                  ) )    .-'       `./ /   \
                 ( (   ,'            `/    /:
                  \ `-"             \'\   / |
                   .              ,  \ \ /  |
                   / @          ,'-`----Y   |
                  (            ;        :   :
                  |  .-.   _,-'         |  /
                  |  | (  (             | /
                  )  (  \  `.___________:/
                  `..'   `--' :mailcat:

Error while checking gmail: timed out
Error while checking yandex: timed out
Error while checking mailDe: timed out
Aikq:
*  test@aikq.pl
*  test@mailbox.info
*  test@mails.info
*  test@aikq.us
Runbox:
*  test@rbx.life
*  test@runbox.com
*  test@runbox.uk
*  test@runbox.us
*  test@xobnur.uk
Fastmail:
*  test@fastmail.se
*  test@fastmail.to
*  test@fastmail.uk
*  test@fastmail.us
*  test@123mail.org
*  test@airpost.net
*  test@mm.st
*  test@myfastmail.com
*  test@proinbox.com
*  test@rushpost.com
*  test@sent.as
*  test@sent.at
*  test@sent.com
*  test@speedymail.org
*  test@warpmail.net
*  test@xsmail.com
*  test@150mail.com
*  test@2-mail.com
*  test@4email.net
*  test@50mail.com
*  test@allmail.net
*  test@bestmail.us
*  test@emailuser.net
*  test@f-m.fm
*  test@fastest.cc
*  test@fea.st
*  test@fmailbox.com
*  test@ftml.net
*  test@imap.cc
*  test@jetemail.net
*  test@mailc.net
*  test@mailftp.com
*  test@mailite.com
*  test@mailnew.com
*  test@mailsent.net
*  test@mailup.net
*  test@mailworks.org
*  test@mymacmail.com
*  test@nospammail.net
*  test@ownmail.net
*  test@petml.com
*  test@realemail.net
*  test@reallyfast.info
*  test@speedpost.net
*  test@ssl-mail.com
*  test@theinternetemail.com
*  test@veryfast.biz
*  test@veryspeedy.net
*  test@yepmail.net
*  test@your-mail.com

**Comment by soxoj** [comment]
And once again with certain platforms, please:
```
python3 ./mailcat.py test -s -p gmail -p yandex -p mailDe
```

**Comment by jasonbjorn01** [comment]
C:\Users\cmder\mailcat-proxy-timeouts-errors
λ python3 ./mailcat.py test -s -p gmail -p yandex -p mailDe
Error while checking gmail: timed out
Error while checking yandex: timed out
Error while checking mailDe: timed out

**Comment by jasonbjorn01** [comment]
And the same while using tor;
λ python3 ./mailcat.py test -s -p gmail -p yandex -p mailDe --tor
Using tor to make requests...
Error while checking gmail: timed out
Error while checking yandex: timed out
Error while checking mailDe: timed out

**Comment by soxoj** [comment]
Hey, check, please, [this version](https://github.com/sharsil/mailcat/archive/refs/heads/aiosmtp-checker.zip) with async SMTP.

**Comment by jasonbjorn01** [comment]
It seems to be missing a module
λ python3 mailcat.py -p gmail  patrickwhite1985 -svd
Traceback (most recent call last):
  File "C:\Users\cmder\mailcat-aiosmtp-checker\mailcat.py", line 10, in <module>
    import aiosmtplib
ModuleNotFoundError: No module named 'aiosmtplib'

**Comment by soxoj** [comment]
Install it, please:
`pip3 install aiosmtplib `

**Comment by jasonbjorn01** [comment]
Installation done.

Here's the output from my new search;
λ python3 mailcat.py -p gmail  patrickwhite1985 -svd
Running gmail checker for patrickwhite1985...
Timed out connecting to alt1.gmail-smtp-in.l.google.com. on port 25
Traceback (most recent call last):
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 1041, in create_connection
    sock = await self._connect_sock(
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 955, in _connect_sock
    await self.sock_connect(sock, address)
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\asyncio\proactor_events.py", line 702, in sock_connect
    return await self._proactor.connect(sock, address)
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\asyncio\tasks.py", line 492, in wait_for
    fut.result()
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\site-packages\aiosmtplib\connection.py", line 366, in _create_connection
    transport, _ = await asyncio.wait_for(connect_coro, timeout=self.timeout)
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\asyncio\tasks.py", line 494, in wait_for
    raise exceptions.TimeoutError() from exc
asyncio.exceptions.TimeoutError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\cmder\mailcat-aiosmtp-checker\mailcat.py", line 87, in code250
    await server.connect(hostname=mxRecord)
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\site-packages\aiosmtplib\connection.py", line 314, in connect
    raise exc
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\site-packages\aiosmtplib\connection.py", line 311, in connect
    response = await self._create_connection()
  File "C:\Users\DE1\AppData\Local\Programs\Python\Python39\lib\site-packages\aiosmtplib\connection.py", line 374, in _create_connection
    raise SMTPConnectTimeoutError(
aiosmtplib.errors.SMTPConnectTimeoutError: Timed out connecting to alt1.gmail-smtp-in.l.google.com. on port 25
No results for gmail
Error while checking gmail: Timed out connecting to alt1.gmail-smtp-in.l.google.com. on port 25

**Comment by suciokhan** [comment]
I'm having similar problems...lots of TimeoutError instances when I try to do a full scan, but it works if I tell it to only look at google with `./mailcat.py -p gmail  username -svd --tor`

**Comment by soxoj** [comment]
@suciokhan @jasonbjorn01 please, check mailcat version from this branch: https://github.com/sharsil/mailcat/tree/async-refactor

**Comment by soxoj** [comment]
Finally fixed in #43.
